pipeline-modules/app-service/azure/go-build.yaml

```diff
@@ -0,0 +1,43 @@
+provisioner: azure
+name: go-build
+version: 1
+revision: 1
+displayName: Go Build
+description: Build source code using Golang
+target: ""
+keywords:
+  - Go
+  - Install
+  - Build
+author: CloudCover
+meta: {}
+inputs:
+  properties:
+    go_build_args:
+      title: Go build args
+      description: Arguments for Go build command
+      type: array
+      default: []
+      items:
+        type: string
+    working_dir:
+      title: Working directory
+      description: The current working directory to execute command
+      type: string
+      default: repo
+    go_version:
+      title: Golang version
+      description: The version of Golang to use
+      type: string
+      default: "1.15.15"
+template: |
+  steps:
+  - script: |
+      echo Installing Golang v{{ go_version }}
+      goenv install {{ go_version }}
```

Comment:
did you test this? I recall that "goenv" happened to be installed on AWS and was recommended way to change go version for Code Build but GCP had a different method. Is the goenv tool on the azure agents?

Reply:
> did you test this? I recall that "goenv" happened to be installed on AWS and was recommended way to change go version for Code Build but GCP had a different method. Is the goenv tool on the azure agents?

I tested this and the job completed successfully despite missing `goenv` on azure agent. Because go versions are preinstalled on the agent. But now when i looked into each job step log i see a failure when we try to use `goenv` . I fixed this now using `Task`